src/averageAges.js

```diff
@@ -14,48 +14,51 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateMenAverageAge(people, century = 0) {
+  const filteredMen = century === 0
+    ? people.filter(person => (
+      person.sex === 'm')
+    )
+    : people.filter(person => (
+      person.sex === 'm' && Math.ceil(person.died / 100) === century
+    ));
+
+  const agesArr = filteredMen.map((man) => man.died - man.born);
+  const sumOfAges = agesArr.reduce((sum, n) => sum + n, 0);
+
+  return sumOfAges / agesArr.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average ave of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find the other who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWoman = withChildren
+    ? people.filter(({ name, sex }) => (
+      people.some(({ mother }) => name === mother)
+      && sex === 'f'
+    ))
+
+    : people.filter(({ sex }) => sex === 'f'
+    );
```

Comment:
Here you can acutally keep it in one string
```suggestion
    : people.filter(({ sex }) => sex === 'f');
```